appserver/tests/application/src/test/java/org/glassfish/main/test/app/deploy/DeployOnVirtualServersTest.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+
+package org.glassfish.main.test.app.deploy;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+
+import org.glassfish.main.itest.tools.asadmin.Asadmin;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.exporter.ZipExporter;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import static java.lang.System.Logger.Level.INFO;
+import static java.lang.System.Logger.Level.WARNING;
+import static org.glassfish.main.itest.tools.GlassFishTestEnvironment.getAsadmin;
+import static org.glassfish.main.itest.tools.asadmin.AsadminResultMatcher.asadminOK;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class DeployOnVirtualServersTest {
+
+    private static final System.Logger LOG = System.getLogger(DeployOnVirtualServersTest.class.getName());
+
+    private static final String APP_NAME = DeployOnVirtualServersTest.class.getSimpleName() + "App";
+
+    private static final Asadmin ASADMIN = getAsadmin();
+
+    private static File warFile;
+
+    @BeforeAll
+    public static void prepareDeployment() throws IOException {
+        // Create virtual servers
+        assertThat(ASADMIN.exec("create-virtual-server", "--hosts", "localhost", "vserver1"), asadminOK());
+        assertThat(ASADMIN.exec("create-virtual-server", "--hosts", "localhost", "vserver2"), asadminOK());
+
+        WebArchive webArchive = ShrinkWrap.create(WebArchive.class)
+            .addClass(SimpleResource.class)
+            .addClass(SimpleApplication.class)
+            .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+
+        LOG.log(INFO, webArchive.toString(true));
+
+        File tempDir = Files.createTempDirectory(APP_NAME).toFile();
```

Comment:
Could we let JUnit handle this and cleanup with https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/io/TempDir.html?

Reply:
Why not? ;) Thank you for advice! Fixed.